src/main/scala/kamon/statsd/StatsDReporter.scala

```diff
@@ -21,102 +21,101 @@ import java.nio.ByteBuffer
+      val packetBuffer = new MetricDataPacketBuffer(config.maxPacketSize, clientChannel, config.agentAddress)
 
-      val bucketData = encodeStatsDTimer(bucket.value, bucket.frequency, metric.unit)
-      packetBuffer.appendMeasurement(keyGenerator.generateKey(metric.name, metric.tags), bucketData)
-    }
+      for (counter <- snapshot.metrics.counters) {
+        packetBuffer.appendMeasurement(keyGenerator.generateKey(counter.name, counter.tags), encodeStatsDCounter(counter.value, counter.unit))
+      }
 
-    packetBuffer.flush()
-  }
+      for (gauge <- snapshot.metrics.gauges) {
+        packetBuffer.appendMeasurement(keyGenerator.generateKey(gauge.name, gauge.tags), encodeStatsDGauge(gauge.value, gauge.unit))
+      }
 
+      for (metric <- snapshot.metrics.histograms ++ snapshot.metrics.rangeSamplers;
+           bucket <- metric.distribution.bucketsIterator) {
 
-  private def encodeStatsDTimer(level: Long, count: Long, unit: MeasurementUnit): String = {
-    val samplingRate: Double = 1D / count
-    s"${scale(level, unit)}|ms${if (samplingRate != 1D) "|@" + samplingRateFormat.format(samplingRate) else ""}"
+        val bucketData = encodeStatsDTimer(bucket.value, bucket.frequency, metric.unit)
+        packetBuffer.appendMeasurement(keyGenerator.generateKey(metric.name, metric.tags), bucketData)
+      }
+
+      packetBuffer.flush()
+    }
   }
 
   private def encodeStatsDCounter(count: Long, unit: MeasurementUnit): String = s"${scale(count, unit)}|c"
 
   private def encodeStatsDGauge(value:Long, unit: MeasurementUnit): String = s"${scale(value, unit)}|g"
 
-  private def scale(value: Long, unit: MeasurementUnit): Double = unit.dimension match {
-    case Time         if unit.magnitude != time.seconds       => MeasurementUnit.scale(value, unit, time.seconds)
-    case Information  if unit.magnitude != information.bytes  => MeasurementUnit.scale(value, unit, information.bytes)
-    case _ => value
+  private def encodeStatsDTimer(level: Long, count: Long, unit: MeasurementUnit): String = {
+    val samplingRate: Double = 1D / count
+    s"${scale(level, unit)}|ms${if (samplingRate != 1D) "|@" + samplingRateFormat.format(samplingRate) else ""}"
   }
 
-  private def readConfiguration(config: Config): Configuration = {
-    val statsDConfig = config.getConfig("kamon.statsd")
-    val agentAddress = new InetSocketAddress(statsDConfig.getString("hostname"), statsDConfig.getInt("port"))
-    val maxPacketSize = statsDConfig.getBytes("max-packet-size")
-    val timeUnit = readTimeUnit(statsDConfig.getString("time-unit"))
-    val informationUnit = readInformationUnit(statsDConfig.getString("information-unit"))
-    val keyGenerator = loadKeyGenerator(statsDConfig.getString("metric-key-generator"), config)
-
-    Configuration(agentAddress, maxPacketSize, timeUnit, informationUnit, keyGenerator)
+  private def scale(value: Long, unit: MeasurementUnit): Double = unit.dimension match {
```

Comment:
`scale` basically hard-codes the units to `seconds` and `bytes` although there are configuration values for that - it should respect the `Configuration.timeUnit` and `Configuration.informationUnit` values, right?

You may have a look at the related PR #3 for reference if you like: https://github.com/kamon-io/kamon-statsd/pull/3/files#diff-01d195608b8240fa1234558b7ea2011cR96

Reply:
This was a very good spot, thanks @kongo2002 for making the pull request. I updated my pull request with the same changes.
I kept only the `reference.conf` file unchanged because I wasn't not sure it was right.
```
# All time values are collected in milliseconds,
```
Are you sure the `kamon-core` collects every things in `milliseconds`?
Again thanks,
Alessandro